src/lib/UniversalDiscovery/Event/Subscriber/RichTextEmbedAllowedContentTypes.php

```diff
@@ -42,16 +42,16 @@ public function __construct(
     /**
      * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
      */
-    private function getAllowedContentTypesIdentifiers(): array
+    private function getAllowedContentTypesIdentifiers(array $contentTypesAllowedViaConfig): ?array
     {
         $access = $this->permissionResolver->hasAccess('content', 'read');
         if (!\is_array($access)) {
-            return [];
+            return count($contentTypesAllowedViaConfig) ? $contentTypesAllowedViaConfig : null;
```

Comment:
Empty `array` is a falsy value, you could omit `count`ing:

```suggestion
            return $contentTypesAllowedViaConfig ?: null;
```

Reply:
There is other problem here. According to `\eZ\Publish\API\Repository\PermissionResolver::hasAccess` documentation `false` mean _no access at all_. So whole statement should be returned executed only if  `$access === true`.